tests/test_registration_and_login.py

```diff
@@ -0,0 +1,180 @@
+import os
+
+import pytest
+from dotenv import load_dotenv
+
+# Import the application and database setup
+from hushline import create_app, db
+from hushline.generate_invite_codes import create_invite_code
+
+# Import models and other modules
+from hushline.model import User
+
+
+@pytest.fixture
+def client():
+    # Load environment variables from .env file or the equivalent
+    load_dotenv("env.sh")
+
+    # Use a simple file name without special characters
+    db_file = "hushline.db"
```

Comment:
This should be randomly generated in `/tmp/` otherwise tests will all act on the same file and each test should be completely isolated. This can be done with the `tmpdir` fixture in pytest.